packages/instant/src/redux/reducer.ts

```diff
@@ -7,13 +7,22 @@ import { AsyncProcessState } from '../types';
 
 import { Action, ActionTypes } from './actions';
 
-export interface State {
+interface BaseState {
     selectedAssetData?: string;
     selectedAssetAmount?: BigNumber;
     selectedAssetBuyState: AsyncProcessState;
     ethUsdPrice?: BigNumber;
     latestBuyQuote?: BuyQuote;
 }
+interface StateWithError extends BaseState {
+    latestError: any;
+    latestErrorDismissed: boolean;
+}
+interface StateWithoutError extends BaseState {
+    latestError: undefined;
+    latestErrorDismissed: undefined;
+}
+export type State = StateWithError | StateWithoutError;
```

Comment:
What if we had two types, `BaseState` that looks like above and `ErrorState` that looks like this:
```typescript
interface ErrorState {
    latestError?: any;
    latestErrorDismissed?: boolean;
}
```

Reply:
I like this idea -- but I think with the change to `Present | Hidden` type which is always present might not be as important